Adds modafinil, a sleep suppressant

yorii opened the pull request:
:cl: Modafinil
add: Adds new medicine chem that suppresses sleep and very lightly reduces stunrates, has a very low metabolic rate which is randomized and a low overdose treshold. Overdosing is a lethal oxyloss unless treated. (With epipen urgently and with charcoal/calomel before it puts you to sleep)
/:cl:

[why]: Because this adds an in my opinion needed effect to chemistry that the bartender gets almost for free. Chem recipe should be pretty balanced and the effect itself is basically the same as drinking cafe latte plus a quarter of what nicotine gives for stunrates.

Requires: #34664

TODO: Replace SetSleeping(0) with Sleep immunity trait.

Do Not Merge.

DaedalusGame committed:
- Fixes smoke and foam having 1/4th of the reagents they should have (yoooo)
- Fixes copy_to runtiming if you transfer from a container that contains nothing directly to a reagents datum
- Removes redundant total_volume check in trans_to
- Fixes cryostylane and pyrosium not heating up people
- Fixes cryostylane cooling beakers to negative kelvins
- Fixes cryostylane cooling people to negative kelvins (stop copypasting code for temperature)

ShizCalev committed:
Fixes some kinetic accelerator runtimes

Buggy123 commented:
:+1: We need more anti-sleep shit aside from straight stimulants.

ExcessiveUseOfCobblestone commented:
@Iamgoofball

yorii commented:
@ExcessiveUseOfCobblestone he already said he liked it.

ExcessiveUseOfCobblestone commented:
can't wait for the lean!

Buggy123 commented:
We have a dedicated anti-sleep chem, now we just need a anti-mute/stutter chem!

Repukan commented:
Just purge it with calomel

Buggy123 commented:
Calomel won't help the tinniest bit with other sources of mute, eg genetics, cult stuns, stings, etc.

ShizCalev committed:
Corrects tile color defines

ShizCalev committed:
Brass skewer runtime fix

ShizCalev committed:
Fixes showhud runtime

ShizCalev committed:
Fixed runtimes

kevinz000 commented:
yet another stacked anti stun is probably bad.

yorii commented:
@kevinz000 so what you're saying is I should make it purely a sleep suppressant?

Nabski89 commented:
Up the overdose amount to 15 and change the metabolism accordingly
OR
Make the overdose results less lethal.

This will prevent it from being a fun death syringe option.

Cyberboss committed:
Merge pull request #34534 from ninjanomnom/redo-fire-priority-defines

Makes fire priority values defines

Cyberboss committed:
Merge pull request #34578 from coiax/lightsout-lightsout

Removes lightsout landmark

Buggy123 committed:
Added observer verb View Gases (#34574)

* Added observer verb View Gases

* Fixed atmospheric report plurality correctness

* Changed mole/s to mol.

tgstation-server committed:
Automatic changelog generation for PR #34574 [ci skip]

Cyberboss committed:
Merge pull request #34591 from XDTM/goodBoy

Fixes slimes attacking people with the slime faction

tgstation-server committed:
Automatic changelog generation for PR #34591 [ci skip]

Cyberboss committed:
Adds byond versions to feedback (#34616)

DaxDupont committed:
Fixes cargo's power usage

CryoSpacewalk committed:
jaunting wraiths should now always be facing the correct direction in their animation and sprite before and after jaunt (#34547)

tgstation-server committed:
Automatic changelog generation for PR #34547 [ci skip]

Cyberboss committed:
Removes duplicate goonchat font-awesome (#34400)

* Removes duplicate goonchat font-awesome

* Fix font awesome path

Cyberboss committed:
Revert "Removes duplicate goonchat font-awesome (#34400)" (#34640)

This reverts commit d013e207dca27044a9a6c48a44e9756515908544.

Cyberboss committed:
Merge pull request #34641 from DaxDupont/belt-power

Fixes conveyor belt power usage.

tgstation-server committed:
Automatic changelog generation for PR #34641 [ci skip]

ChangelingRain committed:
Moths, except I have stolen them and they no longer conflict (#34498)

* Moths, except with less conflicts

* if I remove your wings, will you die?

* it would be extremely painful

tgstation-server committed:
Automatic changelog generation for PR #34498 [ci skip]

Cyberboss committed:
Merge pull request #34501 from AutomaticFrenzy/patch/mapping-scrubbers

Allow mapping non-default scrubbers

Cyberboss committed:
Merge pull request #34551 from MrDoomBringer/DropItLikeItsHottest

fixes supply drop pods being impossible to dissassemble

silicons committed:
You can now buckle people to target stakes (#34514)

* Update target_stake.dm

* Update target_stake.dm

* Update target_stake.dm

Commit:
allows servants to cancel warps more intuitively by clicking on the button again (#34504)

* lets you cancel cc warps

* runtimestation

tgstation-server committed:
Automatic changelog generation for PR #34504 [ci skip]

tgstation-server committed:
Automatic changelog compile, [ci skip]

Cyberboss committed:
Revert "Remove unit tests"

This reverts commit 69e27bb76ff78af47d2c1b4b8faf91d8b5f9b128.

Cyberboss committed:
Adds test run mode. Fails travis if it fails (#34198)

* Adds test run

* Add OVERRIDE_LOG_DIRECTORY_PARAMETER

* Fails travis if a test run fails

* Fix travis looping

* Move the unit tests behind a define

* Fix various test run things

* Remove the DEBUG check because that's just you shooting yourself in the foot

* Adds readability spacing. Makes RunUnitTests a generic proc

* Remove unit tests

tgstation-server committed:
Automatic changelog generation for PR #34198 [ci skip]

tgstation-server committed:
Automatic changelog compile, [ci skip]

Cyberboss committed:
Merge pull request #34620 from ShizCalev/color-correction

Corrects tile color defines

ShizCalev committed:
Fixes spider reagens typo

ShizCalev committed:
. = ..()

ShizCalev committed:
Actually fixes wag emote runtime

ShizCalev committed:
Fixes missing MakeSlipper() args

ExcessiveUseOfCobblestone commented:
Ironically 2 people hitting their epipen on theguy ODing on this would probably do more damage than the original OD.

ExcessiveUseOfCobblestone commented on code/modules/reagents/chemistry/reagents/medicine_reagents.dm:
```diff
@@ -1105,3 +1105,64 @@
+	metabolization_rate = 0.1 * REAGENTS_METABOLISM
+	overdose_threshold = 10 // low overdose treshold due to very low and random metabolism rate
+	taste_description = "salt" // it actually does taste salty
+	var/overdose_progress = 0 // to track overdose progress
```
Wait there's no general system for tiered overdoses?

yorii replied:
I'm as surprised as you are

ExcessiveUseOfCobblestone commented on code/modules/reagents/chemistry/reagents/medicine_reagents.dm:
```diff
@@ -1105,3 +1105,64 @@
+		M.SetSleeping(0, FALSE)
+		M.Jitter(1)
+		metabolization_rate = 0.01 * REAGENTS_METABOLISM * rand(5,20) // randomizes metabolism between 0.02 and 0.08 per tick
+		. = 1
```
`TRUE`?

yorii replied:
Yes, without this weird thing stuff like M.SetSleeping(0, FALSE) will not do anything. It's some sort of mob tick update thingamajig as I understand it.

ExcessiveUseOfCobblestone replied:
you put 1 instead of "TRUE"

ExcessiveUseOfCobblestone commented on code/modules/reagents/chemistry/reagents/medicine_reagents.dm:
```diff
@@ -1105,3 +1105,64 @@
+			M.adjustStaminaLoss(1.5*REM, 0)
+
+	..()
+	. = 1
```
`TRUE`?

yorii replied:
TRUE INDEED

ExcessiveUseOfCobblestone replied:
put "TRUE"

yorii replied:
Why if it's just a boolean? It's literally the same thing and every single other chem in the codebase uses . = 1, should I change all of them?

ExcessiveUseOfCobblestone replied:
If you're using it as T/F it is good code standard to put it as TRUE.

Yes if you'd like.

DaxDupont committed:
Fixes moff wing prefs not being applied

Iamgoofball commented:
hm

@yorii randomize the OD threshold every tick

ShizCalev committed:
Reinvigorates the disco inferno

yorii commented:
@Iamgoofball that's a great fucking idea... I should have thought of that myself

XDTM commented on code/modules/reagents/chemistry/reagents/medicine_reagents.dm:
```diff
@@ -1105,3 +1105,64 @@
+	if(!overdosed) // We do not want any effects on OD
+		M.AdjustStun(-5, 0)
+		M.AdjustKnockdown(-5, 0)
+		M.AdjustUnconscious(-5, 0)
```
this is half a second, intentional?

XDTM commented on code/modules/reagents/chemistry/reagents/medicine_reagents.dm:
```diff
@@ -1105,3 +1105,64 @@
+		M.AdjustKnockdown(-5, 0)
+		M.AdjustUnconscious(-5, 0)
+		M.adjustStaminaLoss(-0.5*REM, 0)
+		M.SetSleeping(0, FALSE)
```
if #34664 is merged you could make a SLEEPIMMUNE trait

yorii replied:
nice, I'll put a reminder down for that 👍

yorii replied:
Actually, SLEEPIMMUNE seems to be sort of required as M.SetSleeping(0, FALSE) doesn't seem to do it, meaning cafe_latte has basically no effect at all on sleeping toxins, only against single one-time sleeps...

ShizCalev committed:
Fixes get_area_name runtime

XDTM commented on code/modules/reagents/chemistry/recipes/medicine.dm:
```diff
@@ -232,4 +232,11 @@
+	id = "modafinil"
+	results = list("modafinil" = 5)
+	required_reagents = list("diethylamine" = 1, "ammonia" = 1, "phenol" = 1, "acetone" = 1, "sulphuric acid" = 1)
+	required_catalysts = list("bromine" = 1) // as close to the real world synthesis as possible
```
maybe that would be a good recipe standard for chems, but right now this would be one of the most complicated chems while being relatively as powerful as coffee and nicotine

yorii replied:
Didn't want to make it too easy though ;p this would pretty much be the only currently existing medical chem right now outside of maybe meth that only has a long-lasting positive buff with little to no downside (if used correctly).

yorii replied:
Nevermind it wasn't actually too hard:
Beaker1:
45 hydrogen
15 nitrogen
15 ethanol
-30 total
Beaker 2:
5 welding fuel
5 carbon
5 hydrogen
-5 total
5 welding fuel
5 oxygen
5 water
5 chlorine
15 Sulphuric acid
1 Bromine
Combine beakers, remove Bromine, done, makes 75u, enough for 10x 7.5u pills (which is only slightly risky). Can also do it via the chemmaster by using the buffer storage to store 1 beaker and also the bromine (which you will need to do anyway because it's a catalyst).

yorii committed:
slightly randomized modafinil OD treshold

yorii committed:
Revert "slightly randomized modafinil OD treshold"

This reverts commit 81814193758ca0b41d96ef0156ce9823868b045b.

yorii committed:
slightly randomized modafinil OD treshold without typo

XDTM committed:
Fixes slimes not attacking anybody (#34667)

tgstation-server committed:
Automatic changelog generation for PR #34667 [ci skip]

lzimann committed:
Merge pull request #34605 from ShizCalev/kinetic_accelerator-runtimes

Fixes some kinetic accelerator runtimes

yorii committed:
sacid typo in recipe

yorii commented:
Doesn't work as intended ;_;
And neither does cafe_latte for that matter, expect something more at a later date.

Commit:
Fixes emergency lights literally never working

MrDoomBringer committed:
Cargotechs now spawn with Export Scanners (#34595)

* excessiveuseof[b]obblestone

* whups

* praisenarsie

tgstation-server committed:
Automatic changelog generation for PR #34595 [ci skip]

ShizCalev committed:
Matched wirecutter colors

MrsTonedOne committed:
Adds nightlies and cdn info

Robustin committed:
Adds the narsour

Robustin committed:
Update drinks_recipes.dm

CryoSpacewalk committed:
fix bananium ore overlay

Cyberboss committed:
Disable ruins for test run mode (#34653)

* Disable ruins for test run mode

* Fix

* Actually, let's enable them all

* That never happened

Tlaltecuhtli committed:
Merge pull request #4 from tgstation/master

test

Tlaltecuhtli committed:
adssdddddddddddddddddddddddd

AnturK committed:
Fixes brother objectives not showing in TP

AnturK committed:
This makes more sense.

Tlaltecuhtli committed:
ugh

Tlaltecuhtli committed:
reverts age 18+ rule

Cyberboss committed:
Merge pull request #34649 from ShizCalev/spider-nutriment-typo

Fixes spider reagents typo

Cyberboss committed:
Merge pull request #34631 from uraniummeltdown/airlock

Emagged Airlocks Are Easier To Deconstruct

tgstation-server committed:
Automatic changelog generation for PR #34631 [ci skip]

ShizCalev committed:
honk

ShizCalev committed:
Unassumes moth genders

tgstation-server committed:
Automatic changelog compile, [ci skip]

Cyberboss committed:
Merge pull request #34654 from ShizCalev/tailwag-fix

Actually fixes wag emote runtime

Cyberboss committed:
Merge pull request #34656 from ShizCalev/makeslippery

Fixes missing MakeSlippery() args

Cyberboss committed:
Removes an unused proc

scrubs2009 committed:
Longer lasting candles.

First time coding anything! 

I was tired of the candles in the chapel going out after ~5 minutes so now they last 5 times longer to make them a viable lighting option

WJohn committed:
Remaps the syndicate ambush space ruin. (#34621)

* Remaps the syndicate ambush space ruin completely.

* conflict fixes

* changes station_lock_override vars to lock_override

* banging donk

tgstation-server committed:
Automatic changelog generation for PR #34621 [ci skip]

AnturK committed:
Fixes tgui reload script (#34632)

* Fixes reload bat

* That's possible

ShizCalev committed:
Stops simple_animals from attacking mobs in godmode

tgstation-server committed:
Automatic changelog compile, [ci skip]

ShizCalev committed:
Fixes bot drop & bot assembly issues

WJohn committed:
gives greyscale sprite much more contrast (#34682)

tgstation-server committed:
Automatic changelog generation for PR #34682 [ci skip]

tgstation-server committed:
Automatic changelog generation for PR #34666 [ci skip]

tgstation-server committed:
Automatic changelog compile, [ci skip]

Robustin committed:
Adds stutter to narsour

tgstation-server committed:
Automatic changelog generation for PR #34485 [ci skip]

DaxDupont committed:
Adds holy vision (Some mobs can now see blessed tiles) (#34497)

* Blessed turfs are visible to people who care about blessed turfs

* A better way

* Adds holy vision

* Compacts special types

* space space space

* Glowing

* Final Solution

* wraith

* bawhoppen has autism and we must hugbox

* Implements feedback

tgstation-server committed:
Automatic changelog generation for PR #34497 [ci skip]

DaedalusGame committed:
- Changes gas reagents (liquid plasma, oxygen, etc) to emit gas based on chem temperature instead of at room temperature when dumped. (#34515)

tgstation-server committed:
Automatic changelog generation for PR #34515 [ci skip]

tgstation-server committed:
Automatic changelog generation for PR #34691 [ci skip]

Lexorion committed:
Adds new eldritch longsword sprite (#34678)

* updates the eldritch longsword sprite

* properly rewinds the left inhand animation

Cyberboss committed:
Readability over micro-optimizations (#34646)

DaxDupont committed:
Makes the lathe repeat it's animation while it's still printing

Cyberboss committed:
Merge pull request #34679 from Xhuis/patch-3

Fixes emergency lights literally never working

tgstation-server committed:
Automatic changelog generation for PR #34679 [ci skip]

Cyberboss committed:
Merge pull request #34692 from Tlaltecuhtli/revs

changes min players for revs to 30 from 20

tgstation-server committed:
Automatic changelog generation for PR #34692 [ci skip]

Cyberboss committed:
Merge pull request #34629 from ShizCalev/showhud-runtime

Fixes ghost show_hud runtime

Cyberboss committed:
Merge pull request #34695 from AnturK/brotherobjectivefix

Fixes brother objectives not showing in TP

Robustin committed:
Ends the great monkey freezing epidemic (#34686)

* Freezing monkey fix

* Update life.dm

* Update life.dm

tgstation-server committed:
Automatic changelog generation for PR #34686 [ci skip]

ShizCalev committed:
current check in ark of the justicar

Cyberboss committed:
Fixes moth heads properly

ShizCalev committed:
Fixes rev mindswap (#34567)

tgstation-server committed:
Automatic changelog generation for PR #34567 [ci skip]

Cyberboss committed:
Config dir again except this time it works (#34606)

* Revert "Revert "Fix config_dir usage (#34469)""

* Make config exist before world/New but not load it until then

ninjanomnom committed:
lets check_dock be silent (#34614)

tgstation-server committed:
Automatic changelog generation for PR #34614 [ci skip]

SpaceManiac committed:
Refactor weather to use Z traits, assorted related cleanup (#34633)

* Add a proc for getting the station center

* Add a couple of comments to ZTRAIT defines

* Remove unused global_map list

* Refactor weather to use the trait system

* Un-hardcode the transit z-level

* Use Z traits to determine Portal Storm event areas

* Fix loading away missions containing anything that reads traits

tgstation-server committed:
Automatic changelog generation for PR #34633 [ci skip]

ShizCalev committed:
Fixes APC controller runtime

tgstation-server committed:
Automatic changelog generation for PR #34717 [ci skip]

SpaceManiac committed:
Fix replace_random_law runtime

KorPhaeron committed:
Rebee atmos (#34723)

MMMiracles committed:
make pretty light go light light (#34720)

tgstation-server committed:
Automatic changelog generation for PR #34720 [ci skip]

deathride58 committed:
removes magic numbers from the set_light var in lights (#34714)

tgstation-server committed:
Automatic changelog generation for PR #34714 [ci skip]

Commit:
Reduces spaces in the Lepidopterian language. (#34712)

tgstation-server committed:
Automatic changelog generation for PR #34712 [ci skip]

Cyberboss committed:
Removes unused consts from dbcore (#34711)

Commit:
Fixes each monkey having it's own individual monkey_team (#34709)

* Fixes monkey teams

* Update monkey.dm

tgstation-server committed:
Automatic changelog generation for PR #34709 [ci skip]

MrsTonedOne committed:
Switches database connector over to mariadb (#34708)

SpaceManiac committed:
Fix radio jammers affecting other z-levels (#34705)

tgstation-server committed:
Automatic changelog generation for PR #34705 [ci skip]

AnturK committed:
Fixes holes in station showing up white on photos. (#34697)

DaxDupont committed:
Adds the camera to the autolathe and changes the materials (#34636)

tgstation-server committed:
Automatic changelog generation for PR #34636 [ci skip]

Cyberboss committed:
Datum var cleanup (#34314)

* Consolidates datum var location

* Move ui_screen to /datum/tgui

* Move focusers to datum.dm. Move fingerprintslast to /atom

* Remove focusers list, makes mobs check if their focus is QDELETED in Life()

* Consolidate use_tag and var_edited into datum_flags

* Revert garbage comment

* Thought I already removed this

* Remove the var_edited preservation

* Removes focus QDELETED handling

ShizCalev committed:
Cleans up unused mob vars

tgstation-server committed:
Automatic changelog compile, [ci skip]

ShizCalev committed:
Corrects addtime stack_trace typo

ShizCalev committed:
Corrects mapGeneratorModule generate() args

ShizCalev committed:
Corrects invalid process_fire args

ShizCalev committed:
Removed named args from visible_message

ShizCalev committed:
Fixes invalid pollCandidatesForMob args

ShizCalev committed:
Fixes syringe icons (#34704)

ShizCalev committed:
Fixes flash_carbon runtime (#34707)

ShizCalev committed:
Adds /area/asteroid/nearstation to blacklist (#34726)

DaxDupont committed:
Fixes no wheat fridge on omega

DaxDupont committed:
this commit kills runtimes

Cyberboss committed:
Cleans up duplicated code and GC issues

WJohn committed:
Tweaks space syndie listening post and lava syndie base. (#34718)

* Tweaks space syndie listening post and lava syndie base.

* Replaces sleeper with medicine locker.

* Gives the syndicate fighter pods and the drop ship the ability to dock at the listening post.

* whoops blacklist

* Removes syndicate access requirements on pirate bridge blast door button.

* Places the airlocks above the firelocks on listening and caravan.

* this kills the dirty vars

* the skill that cuts the DIR and LISTENING vars

tgstation-server committed:
Automatic changelog generation for PR #34718 [ci skip]

Iamgoofball committed:
>people think burn victims are better than literally-cant-see-shit (#34734)

tgstation-server committed:
Automatic changelog generation for PR #34734 [ci skip]

ninjanomnom committed:
Changeling clothes blood fix (#34743)

tgstation-server committed:
Automatic changelog generation for PR #34743 [ci skip]

DaxDupont committed:
Fixes missing title under autolathes (#34753)

tgstation-server committed:
Automatic changelog generation for PR #34753 [ci skip]

vuonojenmustaturska committed:
Replaces a bunch of obj vars (and emagged on machinery/items) with obj_flags (#34078)

* It works, but is it worth it?

* bitfield helpers take 1

* Would this work?

* remove dangling debug code

* rebase & fixes

* vv bitfield stuff, reading

* DNM oceans of shitcode DNM

* honk

* honk2

* plonk

* rebase & fix

tgstation-server committed:
Automatic changelog generation for PR #34078 [ci skip]

Shadowlight213 committed:
Moths now have weak eyes

Shadowlight213 committed:
Fixes no eye damage being done if damage was exactly 3

praisenarsie committed:
removes moth language (#34745)

AnturK committed:
Adds transform helpers to atom vv (#34756)

ShizCalev committed:
martial art sanity check

silicons committed:
Removes pulling claw instant messageless stun (#34742)

tgstation-server committed:
Automatic changelog generation for PR #34742 [ci skip]

Cruix committed:
Consolidated t-ray scanner code (#34759)

* Consolidated t-ray scanner code

* Fixed cables not showing up on t-ray scans

tgstation-server committed:
Automatic changelog generation for PR #34759 [ci skip]

lzimann committed:
Merge pull request #34727 from ShizCalev/simpleanimal-godmode

Stops simple_animals from attacking mobs in godmode

ShizCalev committed:
Fixes java compatability issues

ShizCalev committed:
Fixes portal storm event (#34725)

KorPhaeron committed:
Expands our worst rule

lzimann committed:
Adds the art director as /icons folder codeowner (#34803)

ACCount12 committed:
Makes photocopier use default_unfasten_wrench

KorPhaeron committed:
Fixes being able to lower w_class of transfer valves

tgstation-server committed:
Automatic changelog compile, [ci skip]

optimumtact committed:
Merge pull request #34810 from tgstation/KorPhaeron-patch-2

[s]Fixes being able to lower w_class of transfer valves

silicons committed:
Infrared Refactor (#34142)


bugfix: Infrared beams now update instantly.

tgstation-server committed:
Automatic changelog generation for PR #34142 [ci skip]

MrDoomBringer committed:
Adds the Bluespace Drop Pod upgrade to the cargo express console (#34560)

Researching an upgrade disk from RnD allows you to upgrade the Express Supply Console (#33436).

Instead of normal drop pods, which crash then stick around until you dissassemble them, the console will call down new Bluespace Drop Pods, which will teleport out after crashing (and bring about a slightly smaller crash-explosion too!)

🆑 MrDoomBringer
add: The RnD department can now develop firmware upgrades to the Express Supply Console, unlocking advanced cargo drop pods!
/🆑

why: adds a little bit more techweb-integrated progression to cargo, also part of a compromise from #34374.

oh, also repaths bsdroppod.dm to supplypod.dm

tgstation-server committed:
Automatic changelog generation for PR #34560 [ci skip]

XDTM committed:
Adds a few jelly reagents, small chem adjustments for slimepeople (#34524)

🆑 XDTM
add: Added Regenerative Jelly, made with tricordrazine and slime jelly. It regenerates a bit faster, and won't harm jellypeople.
add: Added Energized Jelly, made with teslium and jelly. It works as an anti-stun chemical for slimepeople and speeds up luminescents' power cooldown, but non-jellypeople will just get shocked.
add: Added Pyroxadone, made with cryoxadone and slime jelly. It's basically inverse cryoxadone: it heals faster the hotter your body temperature is.
tweak: Cryoxadone no longer deals toxin damage to jellypeople.
tweak: Purple Slime Extracts no longer have their sugar->slime jelly reaction (obsolete with extract grinding), and instead have a blood->regen jelly reaction.
tweak: Purple Extract's major activation by Luminescents now give regenerative jelly instead of tricordrazine.
/🆑

Gives a better purpose to purple slimes, makes slime jelly something nice to give to chemistry for improved healing, and makes it so luminescents no longer kill themselves with cryoxadone and tricordrazine.

Also reduced the amount of frost oil given to luminescents in dark blue's activation, so they match cryoxadone's duration.

Fixes #34517
Fixes #34539

tgstation-server committed:
Automatic changelog generation for PR #34524 [ci skip]

MrDoomBringer committed:
Adds ~70 more suicides (#34522)

Does what it says on the tin, probably a terrible idea

tgstation-server committed:
Automatic changelog generation for PR #34522 [ci skip]

coiax committed:
Fake nuclear disks are even more convincing (#34466)

* Fake nuclear disks are even more convincing

:cl: coiax
add: Fake nuclear disks can only be identified by the captain,
observers, nuclear operatives, seeing where the pinpointer points, or
attempting to put it into a nuclear device.
add: Fake nuclear disks "respawn" on the station just like the real one.
/:cl:

- Removes anti-duplication on the syndicate nuke, gets in the way of
testing.

Why? If you manage to pickpocket the Head of Security, steal the disk,
and replace it with a fake, then he shouldn't be able to tell the
difference. Is the disk you're holding real? Maybe you should pop over
to the Vault to check. As a nuke op, leave a fake disk near the very
real, and very armed nuke, as a final taunt to the crew.

tgstation-server committed:
Automatic changelog generation for PR #34466 [ci skip]

optimumtact committed:
Merge pull request #34767 from AutomaticFrenzy/patch/replace-random-law

Fix replace_random_law runtime

tgstation-server committed:
Automatic changelog generation for PR #34767 [ci skip]

optimumtact committed:
Merge pull request #34765 from ShizCalev/apc-controller-runtime

Fixes APC controller runtime

optimumtact committed:
Merge pull request #34750 from ShizCalev/clockwork-current-

current sanity check in ark of the justicar

optimumtact committed:
Merge pull request #34733 from ShizCalev/box-fixes

Fixes bot drop & bot assembly issues

optimumtact committed:
Merge pull request #34804 from tgstation/KorPhaeron-patch-1

Expands our worst rule

Jordie0608 committed:
Expands round end pop recording as json export (#34465)

* expands round end pop recording as json export

* 511 compatibility and fix

* readds feedback for ned

AnturK committed:
Simple rotation component. (#34476)

I'm going to port other rotations to it when i don't feel lazy.
Closes #34064

ninjanomnom committed:
makes the point effect  not trigger movement code

ACCount12 committed:
Plasma cutter now has a tool sound

lzimann committed:
Merge pull request #34778 from ShizCalev/pollCandidatesForMob

Fixes invalid pollCandidatesForMob args

lzimann committed:
Merge pull request #34774 from ShizCalev/addtimer-stack-track-typo

Corrects addtime stack_trace typo

lzimann committed:
Merge pull request #34769 from ShizCalev/var-cleanup

Cleans up unused mob vars

AnturK committed:
Fixes disposal initialization (#34820)

tgstation-server committed:
Automatic changelog compile, [ci skip]

Anonmare committed:
Buffs the Contributing Guide (Fixes a typo)

DaxDupont committed:
Makes the soapstone use examine to use the dialog.

AnturK committed:
Removes animation from photos

tgstation-server committed:
Automatic changelog generation for PR #34795 [ci skip]

YPOQ committed:
Fixes emergency lights having null brightness (#34796)

tgstation-server committed:
Automatic changelog generation for PR #34796 [ci skip]

tgstation-server committed:
Automatic changelog compile, [ci skip]

coiax committed:
Abstract landmarks are now indestructible (#34808)

In situations where a maxcap has been dropped above the Bridge, which is
generally the Observer-Start landmark's location, it would destroy the
landmark, making observers get a debugging message asking them to ahelp
when they observed.

There should be a better way of making an object indestructible from any
ingame effect without having to make it QDEL_HINT_LETMELIVE...

Cyberboss committed:
Merge pull request #34829 from AnturK/photosnotmovies

Removes animation from photos

silicons committed:
Xeno nests are no longer held together by bolts. (#34766)

* Update bed.dm

* Update alien_nest.dm

Cyberboss committed:
Merge pull request #34817 from ninjanomnom/stop-pointing-at-me

Makes the point effect not trigger movement code

tgstation-server committed:
Automatic changelog generation for PR #34817 [ci skip]

Cyberboss committed:
Merge pull request #34781 from DaxDupont/please-make-waffles-for=i-am-belgian

Fixes omega missing a wheat fridge and runtiming pipenets

tgstation-server committed:
Automatic changelog generation for PR #34781 [ci skip]

DaxDupont committed:
Puts beakers in your hand on eject (#34787)

tgstation-server committed:
Automatic changelog generation for PR #34787 [ci skip]

tgstation-server committed:
Automatic changelog compile, [ci skip]

Cyberboss committed:
Merge pull request #34825 from Anonmare/patch-3

Buffs the Contributing Guide (Fixes a typo)

Cyberboss committed:
Add datum flags to the VV helpers (#34822)

lzimann committed:
Merge pull request #34826 from DaxDupont/soapstone

Makes the soapstone engravings use examine to show the dialog

tgstation-server committed:
Automatic changelog generation for PR #34826 [ci skip]

lzimann committed:
Merge pull request #34819 from ACCount12/unwrenching_standardization

Makes photocopier use default_unfasten_wrench

lzimann committed:
Merge pull request #34794 from Shadowlight213/thelightitBURNSSS

Moths now have weak eyes

tgstation-server committed:
Automatic changelog generation for PR #34794 [ci skip]

lzimann committed:
Merge pull request #34777 from ShizCalev/process-fire

Corrects invalid process_fire args

lzimann committed:
Merge pull request #34783 from Cyberboss/FixDumbness

Cleans up duplicated code and GC issues

ShizCalev committed:
Shifts vars off /mob and to the correct subtype (#34773)

lzimann committed:
Merge pull request #34775 from ShizCalev/removed-named-args-from-visible_message

Removed named args from visible_message

tgstation-server committed:
Automatic changelog generation for PR #34690 [ci skip]

KorPhaeron committed:
Fixes the kitchen spike turning people upside down forever (#34807)

XDTM committed:
[Ready]Minor Slime Update: slimepeople get the slime language, slimes speak slime instead of common (#33773)

* Minor Slime Update: slimepeople get the slime language, slimes speak the slime language instead of common

* shadow language

* shadows part 2

tgstation-server committed:
Automatic changelog generation for PR #33773 [ci skip]

Robustin committed:
Gives clockwork marauders a reasonable cost/cooldown (#34285)

balance: Clockwork marauders now take more time (+3s) and power (7x) to create.
balance: The "recent marauder" time limit is now 60 seconds, up from 20. The limit now has a significantly smaller effect on summon time but will act as a further cap on marauder summoning until it has passed.
tweak: The marauder cap will now only account for living cultists.

tgstation-server committed:
Automatic changelog generation for PR #34285 [ci skip]

Commit:
Refactors and nerfs Fleshmend (#34458)

🆑 Xhuis
balance: Fleshmend can no longer be stacked several times at once.
balance: Fleshmend no longer heals changelings who are on fire, and heals burn damage half as quickly as brute and oxygen damage.
code: Fleshmend has been refactored into a status effect.
/🆑

Fleshmend is one of the abilities that makes changelings unkillable combat juggernauts with essentially no counterplay, so I decided to change it:

    First of all, refactored it into a status effect that cannot stack, meaning that only one instance of Fleshmend can be active at once
    Instead of healing 10/10/10 brute/burn/oxy damage, it now heals 10/5/10, emphasizing changelings' weakness to fire
    If a changeling is on fire, Fleshmend will not function at all

This keeps Fleshmend powerful for healing in and out of combat, but adds counterplay to it and gives changelings something to fear: fire. With an extinguisher, they still need to manage their inventory to use it, and even with a fire-immune suit and helmet using the fire on them as a weapon will be unable to use their Fleshmend trump card to outheal any damage they take.

I'll be keeping an eye on changelings after this is merged and adjusting accordingly.

tgstation-server committed:
Automatic changelog generation for PR #34458 [ci skip]

coiax committed:
Costumes for nuclear operatives (#34487)

🆑 coiax
add: Nuclear operatives that wish to fail in style can purchase Centcom
Official and Clown costumes. They can also buy a defective chameleon
kit.
/🆑

It's funny. Other costume ideas welcome.

    Also merges two Initializes in camo guns
    Centcom Inspector outfit does have a Centcom headset, but it has no encryption chips. The outfit also is missing an egun.
    Removes the "Stealthy" description from camo shoes in the nuke op uplink they buy the same damn shoes, they're just more expensive for immersive and plausible reasons.

tgstation-server committed:
Automatic changelog generation for PR #34487 [ci skip]

ExcessiveUseOfCobblestone committed:
One can no longer use cogscarab as a form of Eminence priority. (#34611)

Cogscarabs are meant to build, and wasting an integral (building) slot for something you can get as a ghost so you get priority is shitty to both other ghosts and team (the second one being an admin issue).

If you want to go eminence, you can roll for it as a ghost like everyone else.

lzimann committed:
Merge pull request #34800 from ShizCalev/everybody-was-kungfoo-fighting

martial art sanity check

Ordonis committed:
Adds Ketrazine, a powerful combat stimulant with severe side-effects which can be synthesized by the crew (#34579)

🆑 Ordo
rscadd: Adds Ketrazine, a powerful but dangerous combat stim which the crew can synthesize
/🆑

tgstation-server committed:
Automatic changelog generation for PR #34579 [ci skip]

lzimann committed:
Merge pull request #34801 from ShizCalev/DMM-viewer-fix

Fixes compatability issues with some third-party-java-based™ DMM viewers

lzimann committed:
Merge pull request #34776 from ShizCalev/mapGeneratorModule-args

Corrects mapGeneratorModule generate() args

XDTM committed:
Extends the disability refactor to include more traits, removing some snowflake code (#34664)

This way you can add/remove traits without fear of other sources overriding them. Now you can add TRAIT_STUNIMMUNE to somebody without what if hulk

Notable changes:

    Fakedeath now updates instantly, instead of waiting for the next life tick.
    Fakedeath now sets time of death when acquired.
    Removed extremely snowflake code in reagents that checked if you had morphine to remove slow immunity and so on.
    Hulk no longer overrides status_flag changes, in case there are any.

optimumtact committed:
Merge pull request #34746 from DaxDupont/autolathe

Makes the autolathe repeat it's animation while it's still printing.

tgstation-server committed:
Automatic changelog generation for PR #34746 [ci skip]

Commit:
Shuttles will now throw you when taking off

🆑 SPACE CONDUCTOR
tweak: Shuttles will throw you around if ya don't buckle in
/🆑

tgstation-server committed:
Automatic changelog generation for PR #34744 [ci skip]

optimumtact committed:
Merge pull request #34642 from Cyberboss/UnitTests

Adds unit test framework

optimumtact committed:
Fix old usages of trait introduced by the trait merge

Cyberboss committed:
Merge pull request #34894 from optimumtact/fixmaster

Fix old usages of trait introduced by the trait merge

Cyberboss committed:
Rename /datum/config_entry/var/value to config_entry_value (#34699)

This works around some vv-fu you can use to pass in a config_entry to a proc that reads the value var of some other datum. Byond is stupid enough to actually read it, so it must be uniquely named.

MMMiracles committed:
baseturf editors on syndicate base (#34815)

yorii committed:
Merge branch 'modafinil' of https://github.com/yorii/tgstation into modafinil

yorii commented:
I'm terribly sorry, I did a whole insane mess of things with git... I apologize for this clusterfuck.
